src/dcp/dcp.c

```diff
@@ -271,10 +425,97 @@ int main(int argc, char** argv)
         usage = 1;
     }
 
+    char** argpaths = (&argv[optind]);
+
+#ifdef DAOS_SUPPORT
+    rc = daos_init();
+
+    /* TODO: Don't exit and fail if daos fails init,
+     * could just be regular paths */
+    if (rc != 0) {
+        MFU_LOG(MFU_LOG_ERR, "Failed to initialize daos");
+    }
+
+    struct duns_attr_t dattr  = {0}; 
+    struct duns_attr_t ddattr = {0}; 
+
+    /* Figure out if daos path is the src or dst,
+     * using UNS path, then chop off UNS path
+     * prefix since the path is mapped to the root
+     * of the container in the DAOS DFS mount */
+    if (!daos_uuid_valid(src_pool_uuid) || !daos_uuid_valid(dst_pool_uuid)) {
+        daos_set_paths(&rank, dattr, ddattr, argpaths, &src_pool_uuid, &src_cont_uuid,
+            &dst_pool_uuid, &dst_cont_uuid, dfs_prefix, mfu_src_file, mfu_dst_file);
+    }
+
+    /* check if DAOS source and destination containers are in the same pool */
+    bool same_pool = false;
+    if (mfu_src_file->type == DAOS && mfu_dst_file->type == DAOS) {
+        if (uuid_compare(src_pool_uuid, dst_pool_uuid) == 0) {
+            same_pool = true;
+        }
+    } 
+
+    /* connect to DAOS source pool if uuid is valid */
+    if (mfu_src_file->type == DAOS) {
+       /* if DAOS source pool uuid is valid, then set source file type to DAOS */
+        daos_connect(&rank, &src_poh, &src_coh, &src_pool_uuid, &src_cont_uuid, svc); 
```

Comment:
Do we need error checks on daos_connect?

Reply:
@adammoody the error checking is within the daos_connect function, since it will check for errors when it connects to the pool and opens the container inside the function.

Reply:
What should dcp do if daos_connect fails here?  Do we account for that failure in some other way later on?

Reply:
@adammoody well I've had this happen a few times when I would forgot to start the DAOS servers before running my tests, and DAOS prints and error that says "failed to connect to pool" essentially.

Reply:
It fails within daos when it tries to connect to the pool or open the container, I see what you are saying though, I suppose I can change the error checks in the function to be fatal errors, since I think I have it set to MFU_INFO right now.